Chess/figure.h

```diff
@@ -1,31 +1,30 @@
 #ifndef FIGURE_H
 #define FIGURE_H
 
-#include <QObject>
+#include "common_types.h"
 
 
 ///////////////////////////////////////////////////////////////////////////////////////
 class Figure : public QObject
 {
     Q_OBJECT
-
-private:
-    enum FigType{EMPTY/*base class*/, KING, QUEEN, ROOK, BISHOP, KNIGHT, PAWN};
-    enum FigColor{NONE/*base class*/, WHITE, BLACK};
-
 public:
-    explicit Figure(QObject *parent = 0);
+    friend class Command;
 
-signals:
+    explicit Figure(Desk *parent = 0, FigType ftype=EMPTY, FigColor fcolor=NONE);
 
-public slots:
+    QString getFigName() const;
+    QString getFigColor() const;
 
-private:
-    virtual std::vector<QObject*> cellsToMove();
-    virtual QString figName();
+    //tmp func for Command
+    FigType type();
+    FigColor color();
 
-private:
-    QObject *owner_;    //ptr to figure owner (Desk instance)
+protected:
+    virtual std::vector<Cell> cellsToMove(const Cell &) const;
+
+protected:
+    Desk *owner_;    //ptr to figure owner (Desk instance)

```

Comment:
Using protected is generally a bad idea. It's espesilly bed idea when used for data members. It breaks incapsulation.